CHANGELOG.md

```diff
@@ -5,6 +5,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
```

Comment:
@AVHopp please dont use overall PR comment to ask questions as I cant reply to it in a thread (hence opening here)

- unrelated: I will remove the `read_json` capability again for reason that its not a normal constructor and the functionality I intended will be depretated int he future by pandas. 
- I agree that some user instructions are needed. The whole de/serialization topic is imo worthy of its own user guide, however thats out of scope for this PR. One could add it temporarily somewhere but imo these temp additions are futile

Reply:
Hi guys, just checked my mails and saw this. Will be back this week and can take care of the serialization guide because I already started drafting it before I left 👍

Reply:
Sorry for the wrong comment. Serialization guide would be really great and I agree that this is out of scope for this PR.

Reply:
I've now removed support for `read_json` again. Reasons: i) its not a cosntructor of pd.DataFrame and its not supposed to read json strings (got deprecated future warning)

`from_records` and `from_dict` should be enough to enable suer to input datarames from more userfriendly formats

@AdrianSosic in the modified hook `_structure_dataframe_hook` I thought reusing the `select_constructor_hook` from the same file, but that didnt work immediately out of the box. Didnt investigate further but perhaps the master of the hooks has an immedaite fix for it (or resolve if the current solution is fine)

This PR is not super urgent but should come in a reasonable time frame <2w as BayChem is depending on it for some functionality

Reply:
Can have a look. Can you tell me what exactly you attempted?

Reply:
All good, got in working 👍🏼

Reply:
on core.py, I think instead of 
```
    elif isinstance(obj, dict):
        if constructor := obj.pop("constructor", None):
            # try to get constructor from pandas.DataFrame
            func = getattr(pd.DataFrame, constructor)
            return func(**obj)
        else:
            raise ValueError(
                "For deserializing a dataframe from a dictionary, the 'constructor' "
                "keyword must be provided as key.",
            )
```
something like 
```
    elif isinstance(obj, dict):
        return select_constructor_hook(obj, pd.DataFrame)
```

Reply:
Sorry, probably you didn't see my last message. Already got it working. Will merge the PR 👍🏼